OpenSim/Common/Component.cpp

```diff
@@ -180,7 +180,9 @@ void Component::connect(Component &root)
         AbstractConnector& connector = upd_connectors(ix);
         connector.disconnect();
         try{
-            connector.findAndConnect(root);
+            const Component* connectTo = root.findComponent(connector.get_connected_to_name());
```

Comment:
This was changed because findComponent is a weaker search than findAndConnect which also ensures the connectee is of the correct type and not purely a name match. If anything, `findComponent` should be templated to filter on Component type.